internal/services/mssql/mssql_server_dns_alias_resource.go

```diff
@@ -0,0 +1,138 @@
+package mssql
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ServerDNSAliasModel struct {
+	MsSQLServerId string `tfschema:"mssql_server_id"`
+	Name          string `tfschema:"name"`
+	DNSRecord     string `tfschema:"dns_record"`
+}
+
+type ServerDNSAliasResource struct{}
+
+var _ sdk.Resource = (*ServerDNSAliasResource)(nil)
+
+func (m ServerDNSAliasResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"mssql_server_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ServerID,
+		},
+
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
```

Comment:
Can we set a `ValidateFunc` for this? At least ensure it is not an empty string.

Reply:
added a new validator for it, special characters and very long string are not acceptable by tests